tests/AgentTests/SQLiteTests.hs

```diff
@@ -40,31 +41,46 @@ action `returnsResult` r = runExceptT action `shouldReturn` Right r
 throwsError :: (Eq a, Eq e, Show a, Show e) => ExceptT e IO a -> e -> Expectation
 action `throwsError` e = runExceptT action `shouldReturn` Left e
 
+-- TODO add null port tests
 storeTests :: Spec
 storeTests = withStore do
+  describe "foreign keys enabled" testForeignKeysEnabled
   describe "store methods" do
     describe "createRcvConn" testCreateRcvConn
     describe "createSndConn" testCreateSndConn
-    describe "addSndQueue" testAddSndQueue
-    describe "addRcvQueue" testAddRcvQueue
+    describe "getRcvQueue" testGetRcvQueue
     describe "deleteConn" do
       describe "Receive connection" testDeleteConnReceive
       describe "Send connection" testDeleteConnSend
       describe "Duplex connection" testDeleteConnDuplex
-    describe "Update queue status" do
-      describe "Receive queue" testupdateRcvQueueStatus
-      describe "Send queue" testupdateSndQueueStatus
-      describe "Duplex connection" testUpdateQueueStatusConnDuplex
-      xdescribe "Nonexistent send queue" testUpdateNonexistentSendQueueStatus
-      xdescribe "Nonexistent receive queue" testUpdateNonexistentReceiveQueueStatus
+    describe "upgradeRcvConnToDuplex" testUpgradeRcvConnToDuplex
+    describe "upgradeSndConnToDuplex" testUpgradeSndConnToDuplex
+    describe "Set queue status" do
+      describe "setRcvQueueStatus" testSetRcvQueueStatus
+      describe "setSndQueueStatus" testSetSndQueueStatus
+      describe "Duplex connection" testSetQueueStatusConnDuplex
+      xdescribe "Nonexistent send queue" testSetNonexistentSendQueueStatus
+      xdescribe "Nonexistent receive queue" testSetNonexistentReceiveQueueStatus
     describe "createMsg" do
       describe "A_MSG in RCV direction" testCreateMsgRcv
       describe "A_MSG in SND direction" testCreateMsgSnd
       describe "HELLO message" testCreateMsgHello
       describe "REPLY message" testCreateMsgReply
       describe "Bad queue direction - SND" testCreateMsgBadDirectionSnd
       describe "Bad queue direction - RCV" testCreateMsgBadDirectionRcv
-    describe "getReceiveQueue" testGetReceiveQueue
+
+testForeignKeysEnabled :: SpecWith SQLiteStore
+testForeignKeysEnabled = do
+  it "should throw error if foreign keys are enabled" $ \store -> do
+    let inconsistent_query =
+          [sql|
+            INSERT INTO connections
+              (conn_alias, rcv_host, rcv_port, rcv_id, snd_host, snd_port, snd_id)
+            VALUES
+              ("conn1", "smp.simplex.im", "5223", "1234", "smp.simplex.im", "5223", "2345");
+          |]
+    DB.execute_ (dbConn store) inconsistent_query
+      `shouldThrow` (const True :: Selector DB.SQLError)
```

Comment:
It'd be great if we could check the exact error code here instead of a general exception, but I didn't come up with implementation. Let's discuss if it's easy to add, maybe I'll just add TODO for future.
https://hackage.haskell.org/package/sqlite-simple-0.4.18.0/docs/Database-SQLite-Simple.html#t:SQLError
I believe it should be `ErrorConstraint` in this case